integration/metrics_test.go

```diff
@@ -0,0 +1,221 @@
+	suite.topicData = util.GetLiveMessageTopic(suite.ThingCfg.DeviceID, protocol.TopicAction(dataAction))
+}
+
+func (suite *systemMetricsSuite) TearDownSuite() {
+	suite.SuiteInitializer.TearDown()
+}
+
+func TestSystemMetricsSuite(t *testing.T) {
+	suite.Run(t, new(systemMetricsSuite))
+}
+
+func (suite *systemMetricsSuite) testMetrics(params map[string]interface{}, originators ...string) error {
+	ws, err := util.NewDigitalTwinWSConnection(suite.Cfg)
+	require.NoError(suite.T(), err, "failed to create websocket connection")
+	defer ws.Close()
+
+	err = util.SubscribeForWSMessages(suite.Cfg, ws, "START-SEND-MESSAGES", "")
+	require.NoError(suite.T(), err, "unable to listen for events by using a websocket connection")
+
+	defer func() {
+		_, err := util.ExecuteOperation(suite.Cfg, suite.metricsFeatureURL, "request", map[string]interface{}{
+			"frequency": "0s",
+		})
+		assert.NoError(suite.T(), err, "error while stopping system metrics")
+	}()
+
+	_, err = util.ExecuteOperation(suite.Cfg, suite.metricsFeatureURL, "request", params)
+	require.NoError(suite.T(), err, "error while requesting the system metrics")
+
+	result := util.ProcessWSMessages(suite.Cfg, ws, func(msg *protocol.Envelope) (bool, error) {
+		if msg.Path != suite.pathData {
+			return false, nil
+		}
+
+		if msg.Topic.String() != suite.topicData {
+			return false, nil
+		}
+
+		data, err := json.Marshal(msg.Value)
+		if err != nil {
+			return true, err
+		}
+
+		metric := new(metrics.MetricData)
+		if err := json.Unmarshal(data, metric); err != nil {
+			return true, err
+		}
+
+		var originatorFound bool
+		for _, m := range metric.Snapshot {
+
+			for _, originator := range originators {
+				if originator == m.Originator {
+					originatorFound = true
+				}
+			}
+
+			for _, mm := range m.Measurements {
+				if strings.HasPrefix(mm.ID, "cpu.") {
+					continue
+				}
+
+				if strings.HasPrefix(mm.ID, "memory.") {
+					continue
+				}
+
+				if strings.HasPrefix(mm.ID, "io.") {
+					continue
+				}
+
+				if strings.HasPrefix(mm.ID, "test.") {
+					continue
+				}
+
+				return false, fmt.Errorf("Invalid metrics ID: %s", mm.ID)
+			}
+		}
+
+		return originatorFound, nil
+	})
+
+	return result
+}
+
+func (suite *systemMetricsSuite) TestMetricsRequestDefaultOriginator() {
+	err := suite.testMetrics(map[string]interface{}{
+		"frequency": "5s",
+	}, "SYSTEM")
+	assert.NoError(suite.T(), err, "metrics event from system originator should be received")
+}
+
+func (suite *systemMetricsSuite) TestMetricsRequestMultipleOriginators() {
+	err := suite.testMetrics(map[string]interface{}{
+		"frequency": "5s",
+		"filter": []interface{}{map[string]interface{}{
+			"originator": "SYSTEM",
+		},
+			map[string]interface{}{
+				"id":         []string{"io.*", "cpu.*", "memory.*"},
+				"originator": "suite-connector",
+			},
+		},
+	}, "SYSTEM", "suite-connector")
+	assert.NoError(suite.T(), err, "metrics event from at least one originator system/suite-connector should be received")
```

Comment:
I would suggest to improve the validation and instead of `at least one` to expect all of them.